org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/BNDInstructions.java

```diff
@@ -13,63 +13,46 @@
 package org.eclipse.m2e.pde.target;
 
 import java.io.IOException;
-import java.io.InputStreamReader;
-import java.io.Reader;
 import java.io.StringReader;
-import java.nio.charset.StandardCharsets;
 import java.util.Properties;
 
-import org.apache.commons.io.IOUtils;
+public record BNDInstructions(String key, String instructions) {
 
-public class BNDInstructions {
+	public static final String DEFAULT_INSTRUCTIONS = """
+			Bundle-Name:           Bundle derived from maven artifact ${mvnGroupId}:${mvnArtifactId}:${mvnVersion}
+			version:               ${version_cleanup;${mvnVersion}}
+			Bundle-SymbolicName:   wrapped.${mvnGroupId}.${mvnArtifactId}
+			Bundle-Version:        ${version}
+			Import-Package:        *;resolution:=optional
+			Export-Package:        *;version="${version}";-noimport:=true
+			DynamicImport-Package: *
+			""";
 
-	private static final String BND_DEFAULT_PROPERTIES_PATH = "bnd-default.properties";
 	public static final BNDInstructions EMPTY = new BNDInstructions("", null);
-	private final String key;
-	private final String instructions;
 
 	public BNDInstructions(String key, String instructions) {
 		this.key = key;
 		this.instructions = instructions == null ? null : instructions.strip();
 	}
 
-	public String getKey() {
-		return key;
-	}
-
-	public String getInstructions() {
-		return instructions;
+	public static Properties getDefaultInstructionProperties() {
+		return asProperties(DEFAULT_INSTRUCTIONS);
 	}
 
-	public static BNDInstructions getDefaultInstructions() {
-		try (Reader reader = getDefaultInstructionsReader()) {
-			return new BNDInstructions("", IOUtils.toString(reader));
-		} catch (IOException e) {
-			throw new RuntimeException("load default properties failed", e);
-		}
+	public Properties asProperties() {
+		return asProperties(instructions == null || instructions.isBlank() ? DEFAULT_INSTRUCTIONS : instructions);
 	}
 
-	public Properties asProperties() {
-		Reader reader;
-		if (instructions == null || instructions.isBlank()) {
-			reader = getDefaultInstructionsReader();
-		} else {
-			reader = new StringReader(instructions);
-		}
+	private static Properties asProperties(String instructions) {
 		Properties properties = new Properties();
-		try (reader) {
-			properties.load(reader);
-		} catch (IOException e) {
-			throw new RuntimeException("conversion to properties failed", e);
+		try {
+			properties.load(new StringReader(instructions));
+		} catch (IOException e) { // Cannot happen
```

Comment:
if it cannot happen then an AssertionError should be thrown!

Reply:
Done.